src/main/java/me/lokka30/levelledmobs/LevelledMobs.java

```diff
@@ -117,6 +120,19 @@ public void onEnable() {
             levelManager.startNametagAutoUpdateTask();
             levelManager.startNametagTimer();
         }
+        // Clean up HashMap every 5 minutes
+        hashMapCleanUp=new BukkitRunnable(){
+            @Override
+            public void run() {
+                for(var i:entityDeathInChunkCounter.entrySet()){
+                    var pairList=i.getValue();
+                    while(pairList!=null && !pairList.isEmpty() && Math.abs(pairList.get(0).getKey().getTime()-
+                            System.currentTimeMillis())<=maximumCoolDownTime*1000.0F){
+                        pairList.remove(0);
+                    }
+                }
+            }
+        }.runTaskTimer(this,0,300000);
```

Comment:
How frequently do you want this task to run? This number doesn't seem right to me

`300,000` ticks
`15,000` seconds
`250` minutes
`4.167` hours

Also is it possible we can make it asynchronous? Might need to use a ConcurrentHashMap

Reply:
I thought it’s in milliseconds lol
thank you for the suggestion! Seems to be really good

Reply:
btw, removing first element cost O(n) in arraylist
we should change to other structure(maybe something like std::vector in C++)

Reply:
deque*